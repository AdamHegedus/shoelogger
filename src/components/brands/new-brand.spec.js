import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import NewBrand from '@/components/brands/new-brand';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('new-brand.vue', () => {

    describe('meta.status is false', () => {

        const getters = {
            getMeta: () => {
                return {
                    status: false,
                    message: 'Error message text.'
                };
            }
        };

        const actions = {
            addBrand: jest.fn()
        };

        const store = new Vuex.Store({
            modules: {
                brands: {
                    namespaced: true,
                    getters,
                    actions
                }
            }
        });

        it('should match snapshot when page loaded', () => {
            // ASSIGN
            const wrapper = shallowMount(NewBrand, { store, localVue });

            // ACT
            const actual = wrapper.element;

            // ASSERT
            expect(actual).toMatchSnapshot();

        });

    });

    describe('meta.status is true', () => {

        const getters = {
            getMeta: () => {
                return {
                    status: true,
                    message: null
                };
            }
        };

        const actions = {
            addBrand: jest.fn()
        };

        const store = new Vuex.Store({
            modules: {
                brands: {
                    namespaced: true,
                    getters,
                    actions
                }
            }
        });

        it('should match snapshot when page loaded', () => {
            // ASSIGN
            const wrapper = shallowMount(NewBrand, { store, localVue });

            // ACT
            const actual = wrapper.element;

            // ASSERT
            expect(actual).toMatchSnapshot();

        });

    });

    it('should match snapshot when text input changed', () => {
        // ASSIGN
        const getters = {
            getMeta: () => {
                return {
                    status: true,
                    message: null
                };
            }
        };

        const actions = {
            addBrand: jest.fn()
        };

        const store = new Vuex.Store({
            modules: {
                brands: {
                    namespaced: true,
                    getters,
                    actions
                }
            }
        });
        const wrapper = shallowMount(NewBrand, { store, localVue });
        const expected = 'Test';
        const input = 'Test';

        // ACT
        wrapper.find('input').setValue(input);

        // ASSERT
        const element = wrapper.element;
        const actual = wrapper.vm.$data.brand;
        expect(element).toMatchSnapshot();
        expect(actual).toEqual(expected);

    });

    describe('getBrand', () => {

        it('should return the value of text input', () => {
            // ASSIGN
            const getters = {
                getMeta: () => {
                    return {
                        status: true,
                        message: null
                    };
                }
            };

            const actions = {
                addBrand: jest.fn()
            };

            const store = new Vuex.Store({
                modules: {
                    brands: {
                        namespaced: true,
                        getters,
                        actions
                    }
                }
            });
            const wrapper = shallowMount(NewBrand, { store, localVue });
            const expected = {
                brand: 'Test'
            };
            const input = 'Test';
            wrapper.find('input').setValue(input);

            // ACT
            const actual = wrapper.vm.getBrand;

            // ASSERT
            expect(actual).toEqual(expected);
        });

    });

    describe('addBrand', () => {

        it('should call event with the value of text input', () => {
            // ASSIGN
            const getters = {
                getMeta: () => {
                    return {
                        status: true,
                        message: null
                    };
                }
            };

            const actions = {
                addBrand: jest.fn()
            };

            const store = new Vuex.Store({
                modules: {
                    brands: {
                        namespaced: true,
                        getters,
                        actions
                    }
                }
            });
            const wrapper = shallowMount(NewBrand, { store, localVue });
            const expected = {
                brand: 'Test'
            };
            const input = 'Test';
            wrapper.find('input').setValue(input);

            // ACT
            wrapper.find('button').trigger('click');

            // ASSERT
            expect(actions.addBrand.mock.calls.length).toEqual(1);
            expect(actions.addBrand.mock.calls[0][1]).toEqual(expected);
        });

    });

});
