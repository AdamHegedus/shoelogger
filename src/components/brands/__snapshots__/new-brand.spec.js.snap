// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new-brand.vue meta.status is false should match snapshot when page loaded 1`] = `
<div
  class="container"
>
  <div
    class="row"
  >
    <div
      class="col-sm col-lg-6"
    >
      <h2>
        Add New Brand
      </h2>
       
      <form>
        <div
          class="form-group"
        >
          <label
            for="brandName"
          >
            Brand Name
          </label>
           
          <input
            class="form-control form-control-lg shadow border border-primary"
            id="brandName"
            placeholder="Brand Name"
            type="text"
          />
        </div>
         
        <button
          class="btn btn-primary btn-lg btn-block shadow"
          disabled="disabled"
          type="submit"
        >
          
                    Add
                
        </button>
      </form>
       
      <p
        class="alert alert-danger mt-3"
      >
        
                Error message text.
            
      </p>
    </div>
  </div>
</div>
`;

exports[`new-brand.vue meta.status is true should match snapshot when page loaded 1`] = `
<div
  class="container"
>
  <div
    class="row"
  >
    <div
      class="col-sm col-lg-6"
    >
      <h2>
        Add New Brand
      </h2>
       
      <form>
        <div
          class="form-group"
        >
          <label
            for="brandName"
          >
            Brand Name
          </label>
           
          <input
            class="form-control form-control-lg shadow border border-primary"
            id="brandName"
            placeholder="Brand Name"
            type="text"
          />
        </div>
         
        <button
          class="btn btn-primary btn-lg btn-block shadow"
          disabled="disabled"
          type="submit"
        >
          
                    Add
                
        </button>
      </form>
       
      <!---->
    </div>
  </div>
</div>
`;

exports[`new-brand.vue metaChanged meta.status is true should call event 1`] = `"/brands"`;

exports[`new-brand.vue should match snapshot when text input changed 1`] = `
<div
  class="container"
>
  <div
    class="row"
  >
    <div
      class="col-sm col-lg-6"
    >
      <h2>
        Add New Brand
      </h2>
       
      <form>
        <div
          class="form-group"
        >
          <label
            for="brandName"
          >
            Brand Name
          </label>
           
          <input
            class="form-control form-control-lg shadow border border-primary"
            id="brandName"
            placeholder="Brand Name"
            type="text"
          />
        </div>
         
        <button
          class="btn btn-primary btn-lg btn-block shadow"
          type="submit"
        >
          
                    Add
                
        </button>
      </form>
       
      <!---->
    </div>
  </div>
</div>
`;
